question-servers/freeform.js

```diff
@@ -208,12 +226,17 @@ module.exports = {
                 const courseErr = new Error(htmlFilename + ': ' + err.toString());
                 courseErr.fatal = true;
                 courseErrs.push(courseErr);
-                return callback(null, courseErrs, data, '', fileData);
+                return callback(null, courseErrs, isGetDependencies ? elementDependencies : data, '', fileData);
             }
 
             let index = 0;
             async.eachSeries(elements.keys(), (elementName, callback) => {
                 async.eachSeries($(elementName).toArray(), (element, callback) => {
+                    if (phase === 'get_dependencies' && _.includes(renderedElementNames, element)) {
```

Comment:
Is this just an optimization? I think the dedup code should allow us to unconditionally call `get_dependencies()` repeatedly?